### PR TITLE
#2066-doc-update-layers-v1

### PR DESCRIPTION
**Pull request by nataliyah123**
part of #2066

### CODE REVIEW

File: tensorflow_addons/layers/multihead_attention.py [hunk 1]
```diff
@@ -28,26 +28,23 @@ class MultiHeadAttention(tf.keras.layers.Layer):
     in the tensors `query`, `key`, and `value`, and returns the dot-product attention
     between them:
 
-    ```python
-    mha = MultiHeadAttention(head_size=128, num_heads=12)
+    >>> mha = MultiHeadAttention(head_size=128, num_heads=12)
```
Comment (WindQAQ) [inline tensorflow_addons/layers/multihead_attention.py:32]: Remove this empty line.

File: tensorflow_addons/layers/polynomial.py [hunk 2]
```diff
@@ -40,15 +40,13 @@ class PolynomialCrossing(tf.keras.layers.Layer):
 
     Example:
 
-    ```python
     # after embedding layer in a functional model:
```
Comment (WindQAQ) [inline tensorflow_addons/layers/polynomial.py:43]: Your example does not use functional model. So either remove this line or change your example to functional API.

File: tensorflow_addons/layers/spectral_normalization.py [hunk 3]
```diff
@@ -26,17 +26,14 @@ class SpectralNormalization(tf.keras.layers.Wrapper):
...
-        tf.keras.layers.Dense(n_classes))(net)
-    ```
+    >>> x = tf.random.normal(shape=(1,10,10,1))
+    >>> n_classes = 10
+    >>> net = SpectralNormalization(
+    ... tf.keras.layers.Conv2D(2, 2, activation="relu"),
+    ... input_shape=(10, 10))(x)
+    >>> net = SpectralNormalization(tf.keras.layers.Conv2D(8, 5, activation="relu"))(net)
+    >>> net = SpectralNormalization(tf.keras.layers.Dense(120, activation="relu"))(net)
+    >>> net = SpectralNormalization(tf.keras.layers.Dense(n_classes))(net)
```
Comment (WindQAQ) [inline tensorflow_addons/layers/spectral_normalization.py:36]:
```
    Wrap `tf.keras.layers.Conv2D`:

    >>> x = tf.random.normal(shape=(1, 10, 10, 1))
    >>> conv2d = SpectralNormalization(tf.keras.layers.Conv2D(2, 2))
    >>> y = conv2d(x)
    >>> y.shape

    Wrap `tf.keras.layers.Dense`:

    >>> x = tf.random.normal(shape=(1, 10, 10, 1))
    >>> dense = SpectralNormalization(tf.keras.layers.Dense(10))
    >>> y = dense(x)
    >>> y.shape
```

File: tensorflow_addons/layers/wrappers.py [hunk 4]
```diff
@@ -30,21 +30,18 @@ class WeightNormalization(tf.keras.layers.Wrapper):
...
+    >>> n_classes = 10
+    >>> net = WeightNormalization(
+    ... tf.keras.layers.Conv2D(2, 2, activation='relu'),
+    ... input_shape=input_shape[2:],data_init= False)(x)
+    >>> net = WeightNormalization(
+    ... tf.keras.layers.Conv2D(3, 3, activation='relu'),data_init= False)(net)
+    >>> net = WeightNormalization(
+    ... tf.keras.layers.Dense(4, activation='relu'),data_init= False)(net)
+    >>> net = WeightNormalization(
+    ... tf.keras.layers.Dense(n_classes, activation='softmax'))(net)
```
Comment (WindQAQ) [inline tensorflow_addons/layers/wrappers.py:44]:
```
    Wrap `tf.keras.layers.Conv2D`:

    >>> x = tf.random.normal(shape=(1, 10, 10, 1))
    >>> conv2d = WeightNormalization(tf.keras.layers.Conv2D(2, 2), data_init=False)
    >>> y = conv2d(x)
    >>> y.shape

    Wrap `tf.keras.layers.Dense`:

    >>> x = tf.random.normal(shape=(1, 10, 10, 1))
    >>> dense = WeightNormalization(tf.keras.layers.Dense(10), data_init=False)
    >>> y = dense(x)
    >>> y.shape
```

**Comment by nataliyah123** [comment]
Thanks for the review

File: tensorflow_addons/layers/polynomial.py [hunk 5]
```diff
@@ -40,15 +40,12 @@ class PolynomialCrossing(tf.keras.layers.Layer):
...
-    x2 = PolynomialCrossing(projection_dim=None)((x0, x1))
-    logits = tf.keras.layers.Dense(units=10)(x2)
-    model = tf.keras.Model(input, logits)
-    ```
+    >>> input = np.random.randint(10,size=(10,5))
+    >>> x0 = tf.keras.layers.Embedding(input_dim=10, output_dim=3)(input)
+    >>> x1 = PolynomialCrossing(projection_dim=None)((x0, x0))
+    >>> x2 = PolynomialCrossing(projection_dim=None)((x0, x1))
+    >>> logits = tf.keras.layers.Dense(units=10)(x2)
+    >>> model = tf.keras.Model( logits)
```
Comment (Harsh188) [inline tensorflow_addons/layers/polynomial.py:48]: Could you display the model output here?

**Review by Harsh188 — CHANGES_REQUESTED** [review]

File: tensorflow_addons/layers/polynomial.py (hunk 5 above)
Comment (nataliyah123) [reply tensorflow_addons/layers/polynomial.py:48]: @Harsh188. if you are asking for something like `print(model)` or just `model`  then the answer is no because every time the code runs model object has a different address. 
<tensorflow.python.keras.engine.training.Model at 0x7f4c64f21d68>

File: tensorflow_addons/layers/polynomial.py [hunk 6]
```diff
@@ -40,15 +40,12 @@ class PolynomialCrossing(tf.keras.layers.Layer):
...
-    ```python
-    # after embedding layer in a functional model:
-    input = tf.keras.Input(shape=(None,), name='index', dtype=tf.int64)
-    x0 = tf.keras.layers.Embedding(input_dim=32, output_dim=6))
-    x1 = PolynomialCrossing(projection_dim=None)((x0, x0))
-    x2 = PolynomialCrossing(projection_dim=None)((x0, x1))
-    logits = tf.keras.layers.Dense(units=10)(x2)
-    model = tf.keras.Model(input, logits)
-    ```
+    >>> input = np.random.randint(10,size=(10,5))
```
Comment (WindQAQ) [inline tensorflow_addons/layers/polynomial.py:43]:
```suggestion
    >>> input = np.random.randint(10, size=(10,5))
```

File: tensorflow_addons/layers/polynomial.py (hunk 5 above)
Comment (WindQAQ) [inline tensorflow_addons/layers/polynomial.py:48]:
```suggestion
    >>> model = tf.keras.Model(logits)
```

File: tensorflow_addons/layers/polynomial.py [hunk 7]
```diff
@@ -40,15 +40,12 @@ class PolynomialCrossing(tf.keras.layers.Layer):
...
-    ```python
-    # after embedding layer in a functional model:
-    input = tf.keras.Input(shape=(None,), name='index', dtype=tf.int64)
-    x0 = tf.keras.layers.Embedding(input_dim=32, output_dim=6))
-    x1 = PolynomialCrossing(projection_dim=None)((x0, x0))
-    x2 = PolynomialCrossing(projection_dim=None)((x0, x1))
-    logits = tf.keras.layers.Dense(units=10)(x2)
-    model = tf.keras.Model(input, logits)
-    ```
+    >>> input = np.random.randint(10, size=(10,5))
```
Comment (WindQAQ) [inline tensorflow_addons/layers/polynomial.py:43]:
```suggestion
    >>> input = np.random.randint(10, size=(10, 5))
```

File: tensorflow_addons/layers/multihead_attention.py [hunk 8]
```diff
@@ -28,26 +28,22 @@ class MultiHeadAttention(tf.keras.layers.Layer):
...
-    mha = MultiHeadAttention(head_size=128, num_heads=12)
-
-    query = tf.random.uniform((32, 20, 200)) # (batch_size, query_elements, query_depth)
-    key = tf.random.uniform((32, 15, 300)) # (batch_size, key_elements, key_depth)
-    value = tf.random.uniform((32, 15, 400)) # (batch_size, key_elements, value_depth)
-
-    attention = mha([query, key, value]) # (batch_size, query_elements, value_depth)
-    ```
+    >>> mha = MultiHeadAttention(head_size=128, num_heads=12)
+    >>> query = tf.random.uniform((3, 5, 4)) # (batch_size, query_elements, query_depth)
```
Comment (WindQAQ) [inline tensorflow_addons/layers/multihead_attention.py:32]:
```suggestion
    >>> query = np.random.rand(3, 5, 4) # (batch_size, query_elements, query_depth)
```

File: tensorflow_addons/layers/multihead_attention.py [hunk 9]
```diff
@@ -28,26 +28,22 @@ class MultiHeadAttention(tf.keras.layers.Layer):
...
+    >>> key = tf.random.uniform((3, 6, 5)) # (batch_size, key_elements, key_depth)
```
Comment (WindQAQ) [inline tensorflow_addons/layers/multihead_attention.py:33]:
```suggestion
    >>> key = np.random.rand(3, 6, 5) # (batch_size, key_elements, key_depth)
```

File: tensorflow_addons/layers/multihead_attention.py [hunk 10]
```diff
@@ -28,26 +28,22 @@ class MultiHeadAttention(tf.keras.layers.Layer):
...
+    >>> value = tf.random.uniform((3, 6, 6)) # (batch_size, key_elements, value_depth)
```
Comment (WindQAQ) [inline tensorflow_addons/layers/multihead_attention.py:34]:
```suggestion
    >>> value = np.random.rand(3, 6, 6) # (batch_size, key_elements, value_depth)
```

File: tensorflow_addons/layers/multihead_attention.py [hunk 11]
```diff
@@ -28,26 +28,22 @@ class MultiHeadAttention(tf.keras.layers.Layer):
...
-    ```python
-    mha = MultiHeadAttention(head_size=128, num_heads=12)
-
-    query = tf.random.uniform((32, 20, 200)) # (batch_size, query_elements, query_depth)
-    key = tf.random.uniform((32, 15, 300)) # (batch_size, key_elements, key_depth)
-
-    attention = mha([query, key]) # (batch_size, query_elements, key_depth)
-    ```
+    >>> mha = MultiHeadAttention(head_size=128, num_heads=12)
+    >>> query = tf.random.uniform((3, 5, 5)) # (batch_size, query_elements, query_depth)
```
Comment (WindQAQ) [inline tensorflow_addons/layers/multihead_attention.py:42]:
```suggestion
    >>> query = np.random.rand(3, 5, 5) # (batch_size, query_elements, query_depth)
```

File: tensorflow_addons/layers/multihead_attention.py [hunk 12]
```diff
@@ -28,26 +28,22 @@ class MultiHeadAttention(tf.keras.layers.Layer):
...
+    >>> key = tf.random.uniform((3, 6, 10)) # (batch_size, key_elements, key_depth)
```
Comment (WindQAQ) [inline tensorflow_addons/layers/multihead_attention.py:43]:
```suggestion
    >>> key = np.random.rand(3, 6, 10) # (batch_size, key_elements, key_depth)
```

File: tensorflow_addons/layers/spectral_normalization.py [hunk 13]
```diff
@@ -26,17 +26,21 @@ class SpectralNormalization(tf.keras.layers.Wrapper):
...
-    net = SpectralNormalization(
-        tf.keras.layers.Conv2D(16, 5, activation="relu"))(net)
-    net = SpectralNormalization(
-        tf.keras.layers.Dense(120, activation="relu"))(net)
-    net = SpectralNormalization(
-        tf.keras.layers.Dense(n_classes))(net)
-    ```
+    Wrap `tf.keras.layers.Conv2D`:
+
+    >>> x = tf.random.normal(shape=(1, 10, 10, 1))
```
Comment (WindQAQ) [inline tensorflow_addons/layers/spectral_normalization.py:31]:
```suggestion
    >>> x = np.random.rand(1, 10, 10, 1)
```

File: tensorflow_addons/layers/spectral_normalization.py [hunk 14]
```diff
@@ -26,17 +26,21 @@ class SpectralNormalization(tf.keras.layers.Wrapper):
...
+    >>> conv2d = SpectralNormalization(tf.keras.layers.Conv2D(2, 2))
+    >>> y = conv2d(x)
+    >>> y.shape
+    TensorShape([1, 9, 9, 2])
+
+    Wrap `tf.keras.layers.Dense`:
+
+    >>> x = tf.random.normal(shape=(1, 10, 10, 1))
```
Comment (WindQAQ) [inline tensorflow_addons/layers/spectral_normalization.py:39]:
```suggestion
    >>> x = np.random.rand(1, 10, 10, 1)
```

File: tensorflow_addons/layers/wrappers.py [hunk 15]
```diff
@@ -30,21 +30,21 @@ class WeightNormalization(tf.keras.layers.Wrapper):
...
-    net = WeightNormalization(
-        tf.keras.layers.Dense(120, activation='relu'),
-        data_init=True)(net)
-    net = WeightNormalization(
-        tf.keras.layers.Dense(n_classes),
-        data_init=True)(net)
-    ```
+    Wrap `tf.keras.layers.Conv2D`:
+
+    >>> x = tf.random.normal(shape=(1, 10, 10, 1))
```
Comment (WindQAQ) [inline tensorflow_addons/layers/wrappers.py:35]:
```suggestion
    >>> x = np.random.rand(1, 10, 10, 1)
```

File: tensorflow_addons/layers/wrappers.py [hunk 16]
```diff
@@ -30,21 +30,21 @@ class WeightNormalization(tf.keras.layers.Wrapper):
...
+    >>> conv2d = WeightNormalization(tf.keras.layers.Conv2D(2, 2), data_init=False)
+    >>> y = conv2d(x)
+    >>> y.shape
+    TensorShape([1, 9, 9, 2])
+
+    Wrap `tf.keras.layers.Dense`:
+
+    >>> x = tf.random.normal(shape=(1, 10, 10, 1))
```
Comment (WindQAQ) [inline tensorflow_addons/layers/wrappers.py:43]:
```suggestion
    >>> x = np.random.rand(1, 10, 10, 1)
```

**Review by WindQAQ — APPROVED** [review]
